README.md

```diff
@@ -28,54 +28,47 @@ Require the library
 require 'graphql/batch'
 ```
 
-Define a GraphQL::Batch::Query derived class. Use group_key to specify which queries can be reduced into a batch query, and an execute class method which makes that batch query and passes the result to each individual query.
+Define a custom loader, which is initialized with arguments that are used for grouping and an execute method for doing the batch query. The load method can be overriden to coerce the type of the key.
 
 ```ruby
-class FindQuery < GraphQL::Batch::Query
-  attr_reader :model, :id
+class RecordLoader < GraphQL::Batch::Loader
+  attr_reader :model
 
-  def initialize(model, id, &block)
+  def initialize(model)
+    super()
     @model = model
-    @id = id
-    super(&block)
   end
 
-  # super returns the class name
-  def group_key
-    "#{super}:#{model.name}"
+  def load(key)
+    super(Integer(key))
   end
 
-  def self.execute(queries)
-    model = queries.first.model
-    ids = queries.map(&:id)
-    records_by_id = model.where(id: ids).index_by(&:id)
-    queries.each do |query|
-      query.complete(records_by_id[query.id])
-    end
+  def execute
+    model.find(keys).each { |record| fulfill(record.id, record) }
   end
 end
 ```
 
-When defining your schema, using the graphql gem, return a your batch query object from the resolve proc.
+Use the batch execution strategy with your schema
 
 ```ruby
-resolve -> (obj, args, context) { FindQuery.new(Product, args["id"]) }
+MySchema = GraphQL::Schema.new(query: MyQueryType)
+MySchema.query_execution_strategy = GraphQL::Batch::ExecutionStrategy
 ```
 
-Use the batch execution strategy with your schema
+When defining your schema, using the graphql gem, return the result of calling `.load` on your loader object from the resolve proc.
```

Comment:
Also need to document use of `for` and the group keys instead of instantiating a new one